src/Package/Impl/ToolWindows/PlotDeviceWindowPane.cs

```diff
@@ -0,0 +1,88 @@
+using Microsoft.R.Components.Plots.Implementation.ViewModel;
+using Microsoft.R.Components.Settings;
+using Microsoft.R.Host.Client;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.R.Package.Commands;
+using Microsoft.VisualStudio.R.Package.Interop;
+using Microsoft.VisualStudio.R.Package.Shell;
+using Microsoft.VisualStudio.R.Packages.R;
+
+namespace Microsoft.VisualStudio.R.Package.ToolWindows {
+    [Guid(WindowGuidString)]
+    internal class PlotDeviceWindowPane : VisualComponentToolWindow<IRPlotDeviceVisualComponent>, IOleCommandTarget {
+        private readonly IRPlotManager _plotManager;
+        private readonly IRSession _session;
+        private readonly IRSettings _settings;
+        private readonly ICoreShell _coreShell;
+        private readonly int _instanceId;
+        private IOleCommandTarget _commandTarget;
+
+        public const string WindowGuidString = "993DB88E-ED56-4503-9704-CA90EDAE505C";
+
+        public static Guid WindowGuid { get; } = new Guid(WindowGuidString);
+
+        public PlotDeviceWindowPane(IRPlotManager plotManager, IRSession session, int instanceId, IRSettings settings, ICoreShell coreShell) {
+            _plotManager = plotManager;
+            _session = session;
+            _instanceId = instanceId;
+            _settings = settings;
+            _coreShell = coreShell;
+
+            // this value matches with icmdShowPlotWindow's Icon in VSCT file
+            BitmapImageMoniker = KnownMonikers.LineChart;
+            Caption = Resources.PlotWindowCaption;
+            ToolBar = new CommandID(RGuidList.RCmdSetGuid, RPackageCommandId.plotWindowToolBarId);
+        }
+
+        protected override void OnCreate() {
+            var controller = new AsyncCommandController();
+            var viewModel = new RPlotDeviceViewModel(_plotManager, _session, _instanceId);
+            //controller.AddCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdNewPlotWindow, PlotDeviceCommandFactory.NewPlotDevice(_plotManager.InteractiveWorkflow));
+            controller.AddCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdActivatePlotWindow, PlotDeviceCommandFactory.ActivatePlotDevice(_plotManager.InteractiveWorkflow, viewModel));
+            controller.AddCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdClearPlots, PlotDeviceCommandFactory.RemoveAllPlots(_plotManager.InteractiveWorkflow, viewModel));
+            controller.AddCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdRemovePlot, PlotDeviceCommandFactory.RemoveCurrentPlot(_plotManager.InteractiveWorkflow, viewModel));
+            controller.AddCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdNextPlot, PlotDeviceCommandFactory.NextPlot(_plotManager.InteractiveWorkflow, viewModel));
+            controller.AddCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdPrevPlot, PlotDeviceCommandFactory.PreviousPlot(_plotManager.InteractiveWorkflow, viewModel));
```

Comment:
- Let's make `PlotDeviceCommandFactory` an instance. You'll be able to pass `InteractiveWorkflow` and `viewModel` to its ctor and avoid providing them for every command
- We have `CommandAsyncToOleMenuCommandShimFactory` that with standard guids. We probably can add an extension method that will do the same